src/userflow.ts

```diff
@@ -206,6 +207,30 @@ interface Deferred {
 // empty object instead of `window`.
 var w: WindowWithUserflow = typeof window === 'undefined' ? ({} as any) : window
 var userflow = w.userflow
+
+// indicates if the history API's pushState and replaceState are patched with custom event emitters
+w.__userflowStatePatched = false
+
+// patch the history API's
+// pushState method to emit userflow:pushstate and
+// replaceState method to emit userflow:replacestate
+const originalPushState = window.history.pushState
+const originalReplaceState = window.history.replaceState
+
+w.history.pushState = (...args) => {
+  const userflowPushStateEvent = new CustomEvent('userflow:pushstate')
+  w.__userflowStatePatched = true
+  document.dispatchEvent(userflowPushStateEvent)
+  originalPushState.apply(w.history, args)
+}
```

Comment:
Should return the value from the original method. I know it's "normally" `undefined`, but it seems good practice to return the original value when override a native function.